cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -131,6 +134,38 @@ func getKubeEnvValueFromTemplateMetadata(template *gce.InstanceTemplate) (string
 	return "", nil
 }
 
+// BuildAutoscalingOptions builds config.NodeGroupAutoscalingOptions from the provided GCE template.
+func (t *GceTemplateBuilder) BuildAutoscalingOptions(template *gce.InstanceTemplate, defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
+	if template.Properties == nil {
+		return nil, fmt.Errorf("instance template %s has no properties", template.Name)
+	}
+
+	kubeEnvValue, err := getKubeEnvValueFromTemplateMetadata(template)
+	if err != nil {
+		return nil, fmt.Errorf("could not obtain kube-env from template metadata; %v", err)
+	}
+
+	options, err := extractAutoscalingOptionsFromKubeEnv(kubeEnvValue)
+	if err != nil {
+		return nil, err
+	}
+
+	if opt, ok := getFloat64Option(options, template.Name, "scaledownutilizationthreshold"); ok {
```

Comment:
These strings occur in other cloud provider PRs as well. Is it possible to make them global constants?

Reply:
Given the way to pass the config is different between providers do we really see value in shared string consts? I can easily see providers striving to make them consistent with the rest of config for given provider (ex. use the same prefix), so we can't guarantee they would be portable anyway. 

That being said I would also prefer if the strings were consts instead (whether defined at provider level or as something like DefaultScaleDownUtilizationThresholdKey next to NodeGroupAutoscalingOptions definition).

Reply:
Currently we have in-code constants that the author defined (I assume) that happen to be exactly the same as on other cloud providers. We could try to push it more towards a standard. 
But if you prefer local constants then I'm fine with this solution as well. It will still be better than magical strings hidden in code :).

Reply:
Made that shared constants (we don't often have opportunities to enforce some consistency ;), and changed AWS and Azure PRs to draft until this gets in (so I'll refresh them to pick those symbols + feedback from this PR).